src/AggregateRoots/FakeAggregateRoot.php

```diff
@@ -36,7 +38,18 @@ public function given($events): self
 
     public function when($callable): self
     {
-        $callable($this->aggregateRoot);
+        $this->whenResult = $callable($this->aggregateRoot);
+
+        return $this;
+    }
+
+    public function assertThat(callable $callback): self
+    {
+        $result = $callback($this->whenResult);
+
+        if ($result !== null) {
```

Comment:
I explicitly only assert when a result is returned. If no result is returned and no other assertions are performed — with `assertThat` or another assert function — PHPUnit will warn you that this test case is not doing any assertions. I think this is better than having the `$callback` result default to `true`.